app/actions/remote/channel.ts

```diff
@@ -49,6 +51,96 @@ export type MyChannelsRequest = {
     error?: unknown;
 }
 
+export type ChannelMembersRequest = {
+    members?: ChannelMembership[];
+    error?: unknown;
+}
+
+export async function removeMemberFromChannel(serverUrl: string, channelId: string, userId: string) {
+    try {
+        const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+        const client = NetworkManager.getClient(serverUrl);
+
+        await client.removeFromChannel(userId, channelId);
+        await deleteChannelMembership(operator, userId, channelId);
+
+        return {error: undefined};
+    } catch (error) {
+        forceLogoutIfNecessary(serverUrl, error as ClientErrorProps);
+        return {error};
+    }
+}
+
+export async function fetchChannelMembersByIds(serverUrl: string, channelId: string, userIds: string[], fetchOnly = false): Promise<ChannelMembersRequest> {
+    try {
+        const client = NetworkManager.getClient(serverUrl);
+        const members = await client.getChannelMembersByIds(channelId, userIds);
+
+        if (!fetchOnly) {
+            const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
+            if (operator && members.length) {
+                const modelPromises: Array<Promise<Model[]>> = [];
+                const membership = members.map((u) => ({
+                    channel_id: channelId,
+                    user_id: u.user_id,
+                    scheme_admin: u.scheme_admin,
+                }));
+                modelPromises.push(operator.handleChannelMembership({
+                    channelMemberships: membership,
+                    prepareRecordsOnly: true,
+                }));
+
+                const models = await Promise.all(modelPromises);
+                await operator.batchRecords(models.flat());
+            }
+        }
+
+        return {members};
+    } catch (error) {
+        forceLogoutIfNecessary(serverUrl, error as ClientError);
+        return {error};
+    }
+}
+export async function updateChannelMemberSchemeRoles(serverUrl: string, channelId: string, userId: string, isSchemeUser: boolean, isSchemeAdmin: boolean) {
```

Comment:
Shouldn't this add the possibility of also update the database? Or that kind of update is done elsewhere?

Reply:
Yes, this needs to be added. I created a ticket to add it. https://mattermost.atlassian.net/browse/MM-49672

Reply:
fetchonly added as an option